src/datachain/lib/dc/datachain.py

```diff
@@ -176,6 +202,16 @@ def __repr__(self) -> str:
         self.print_schema(file=file)
         return file.getvalue()
 
+    def _as_delta(self, delta: bool = False) -> "Self":
+        """Marks this chain as delta, which means special delta process will be
+        called on saving dataset for optimization"""
+        self._delta = delta
+        return self
+
+    @property
+    def delta(self) -> bool:
```

Comment:
is it public? do we want it to be public?

Reply:
no, that's fine .. but if we keep it public we need proper docs for it then ... and an example if you have something in mind

Reply:
one minor things that is left here @ilongin  ... let's please take care of it